velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -68,20 +68,69 @@ void WindowFuzzer::addWindowFunctionSignatures(
   }
 }
 
-std::tuple<std::string, bool> WindowFuzzer::generateFrameClause() {
-  auto frameType = [](int value) -> const std::string {
+void WindowFuzzer::addKRangeFrameColumnToInput(
+    std::vector<RowVectorPtr>& input,
+    core::WindowNode::BoundType& frameBoundType,
+    const std::string& columnName) {
+  // For frames with k RANGE PRECEDING/FOLLOWING, Velox requires the application
+  // to add columns with the range frame boundary value computed according
+  // to the frame type.
+  // If the frame is k PRECEDING :
+  // frame_boundary_value = current_order_by - k (for ascending ORDER BY)
+  // frame_boundary_value = current_order_by + k (for descending ORDER BY)
+  // If the frame is k FOLLOWING :
+  // frame_boundary_value = current_order_by + k (for ascending ORDER BY)
+  // frame_boundary_value = current_order_by - k (for descending ORDER BY)
+  auto computeOffset = [&](const core::WindowNode::BoundType& frameBoundType,
+                           const int64_t& offsetValue,
+                           const bool& ascending) -> int64_t {
+    if (frameBoundType == core::WindowNode::BoundType::kCurrentRow) {
+      return 0;
+    }
+    if (frameBoundType == core::WindowNode::BoundType::kPreceding) {
+      return ascending ? -offsetValue : offsetValue;
+    } else {
+      return ascending ? offsetValue : -offsetValue;
+    }
+  };
+
+  auto size = vectorFuzzer_.getOptions().vectorSize;
+  velox::test::VectorMaker vectorMaker{pool_.get()};
+
+  for (auto j = 0; j < FLAGS_num_batches; ++j) {
+    int64_t offsetValue =
```

Comment:
We shouldn't be changing offset per batch. There should only be a single offset for the entire input.